doc/library/sandbox/cuda/dnn.txt

```diff
@@ -145,42 +145,68 @@ get an error when cuDNN can not be used with them, use this flag:
     cudnn.h must be readable by everybody.
 
 
-Functions
-=========
+* :ref:`Convolution <libdoc_cuda_dnn_convolution>`
+* :ref:`Pooling <libdoc_cuda_dnn_pooling>`
+* :ref:`Batch Normalization <libdoc_cuda_dnn_bn>`
+* :ref:`RNN <libdoc_cuda_dnn_rnn>`
+* :ref:`Softmax <libdoc_cuda_dnn_softmax>`
+* :ref:`Internal Ops <libdoc_cuda_dnn_internal_ops>`
+
+.. _libdoc_cuda_dnn_convolution:
+
+Convolution
+===========
 
 .. automodule:: theano.sandbox.cuda.dnn
    :noindex:
-   :members: dnn_conv, dnn_pool, dnn_conv3d, dnn_gradweight, dnn_gradinput, dnn_pool, dnn_batch_normalization_train, dnn_batch_normalization_test
+   :members: dnn_conv, dnn_conv3d, dnn_gradweight, dnn_gradinput
 
-Convolution Ops
-===============
+.. _libdoc_cuda_dnn_pooling:
+
+Pooling
+=======
 
 .. automodule:: theano.sandbox.cuda.dnn
    :noindex:
-   :members: GpuDnnConvDesc, GpuDnnConv, GpuDnnConv3d, GpuDnnConvGradW,
-             GpuDnnConv3dGradW, GpuDnnConvGradI, GpuDnnConv3dGradI
+   :members: dnn_pool
 
-Pooling Ops
-===========
+.. _libdoc_cuda_dnn_bn:
+
+Batch Normalization
+===================
 
 .. automodule:: theano.sandbox.cuda.dnn
    :noindex:
-   :members: GpuDnnPoolDesc, GpuDnnPool, GpuDnnPoolGrad
+   :members: dnn_batch_normalization_train, dnn_batch_normalization_test
 
-Softmax Ops
-===========
+.. _libdoc_cuda_dnn_rnn:
 
-.. automodule:: theano.sandbox.cuda.dnn
+RNN
+===
+
+New back-end only! Without dropout support.
+
+.. automodule:: theano.gpuarray.dnn
    :noindex:
-   :members: GpuDnnSoftmax, GpuDnnSoftmaxGrad
+   :members: RNNBlock
```

Comment:
This should also be added to the CuDNN section of the theano.gpuarray.dnn documentation.